pkg/apis/bookkeeper/v1alpha1/status_test.go

```diff
@@ -72,13 +141,170 @@ var _ = Describe("BookkeeperCluster Status", func() {
+			BeforeEach(func() {
+				bk.Status.SetErrorConditionFalse()
+				bk.Status.SetErrorConditionTrue("RollbackFailed", " ")
+			})
+			It("should return rollback failed state to true using function", func() {
+				Ω(bk.Status.IsClusterInRollbackFailedState()).To(Equal(true))
+			})
+			It("should return rollback failed state to false using function", func() {
+				bk.Status.SetErrorConditionTrue("some err", "")
+				Ω(bk.Status.IsClusterInRollbackFailedState()).To(Equal(false))
+			})
+
+			It("should have pods Error condition with true status using function", func() {
+				Ω(bk.Status.IsClusterInErrorState()).To(Equal(true))
+			})
+			It("should have pods Error condition with true status", func() {
+				_, condition := bk.Status.GetClusterCondition(v1alpha1.ClusterConditionError)
+				Ω(condition.Status).To(Equal(corev1.ConditionTrue))
+
+			})
+		})
+		Context("set pod Error condition to be false", func() {
+			BeforeEach(func() {
+				bk.Status.SetErrorConditionTrue("UpgradeFailed", " ")
+				bk.Status.SetErrorConditionFalse()
+			})
+
+			It("should have Error condition with false status", func() {
+				_, condition := bk.Status.GetClusterCondition(v1alpha1.ClusterConditionError)
+				Ω(condition.Status).To(Equal(corev1.ConditionFalse))
+			})
+
+			It("should have Error condition with false status using function", func() {
+				Ω(bk.Status.IsClusterInUpgradeFailedState()).To(Equal(false))
+			})
+
+			It("cluster in error state should return false", func() {
+				Ω(bk.Status.IsClusterInErrorState()).To(Equal(false))
+			})
+
+			It("should have updated timestamps", func() {
+				_, condition := bk.Status.GetClusterCondition(v1alpha1.ClusterConditionError)
+				//check the timestamps
+				Ω(condition.LastUpdateTime).NotTo(Equal(""))
+				Ω(condition.LastTransitionTime).NotTo(Equal(""))
+			})
+		})
+		Context("set pods rollback condition to be true", func() {
+			BeforeEach(func() {
+				bk.Status.SetRollbackConditionFalse()
+				bk.Status.SetRollbackConditionTrue(" ", " ")
+				bk.Status.UpdateProgress("UpgradeErrorReason", "")
+			})
+			It("should have pods rollback condition with true status", func() {
+				_, condition := bk.Status.GetClusterCondition(v1alpha1.ClusterConditionRollback)
+				Ω(condition.Status).To(Equal(corev1.ConditionTrue))
+			})
+			It("should have pods rollback condition with true status using function", func() {
+				Ω(bk.Status.IsClusterInRollbackState()).To(Equal(true))
+			})
+			It("Checking GetlastCondition function and It should return RollbackCondition as cluster in Rollback state", func() {
+				condition := bk.Status.GetLastCondition()
+				Ω(string(condition.Type)).To(Equal(v1alpha1.ClusterConditionRollback))
+			})
+			It("Checking ClusterInUpgradeFailedOrRollbackState and It should return true", func() {
+				Ω(bk.Status.IsClusterInUpgradeFailedOrRollbackState()).To(Equal(true))
+			})
+
+			Context("set pods rollback condition to be false", func() {
```

Comment:
Not aligned similar to the other Contexts

Reply:
Fixed